jerry-core/ecma/builtin-objects/ecma-builtin-arraybuffer-prototype.c

```diff
@@ -227,6 +242,36 @@ ecma_builtin_arraybuffer_prototype_object_slice (ecma_value_t this_arg, /**< thi
   return ret_value;
 } /* ecma_builtin_arraybuffer_prototype_object_slice */
 
+/**
+ * Dispatcher of the built-in's routines
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+ecma_value_t
+ecma_builtin_arraybuffer_prototype_dispatch_routine (uint8_t builtin_routine_id, /**< built-in routine identifier */
+                                                     ecma_value_t this_arg, /**< 'this' argument value */
+                                                     const ecma_value_t arguments_list_p[], /**< list of arguments
+                                                                                            *   passed to routine */
+                                                     uint32_t arguments_number) /**< length of arguments' list */
+{
```

Comment:
The checks from 108-119 lines (numbering based on the new line counts) seems to be similar in both methods. AFAIK we could extract those changes to this place, just before the switch.

Reply:
I second, move the isArrayBuffer validation from the methods to the dispatcher.

Reply:
This seems still mising.